dashboard/client/src/components/ActorTable.component.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { render } from "@testing-library/react";
+import React from "react";
+import { MemoryRouter } from "react-router-dom";
+import { Actor } from "../type/actor";
+import ActorTable from "./ActorTable";
+describe("ActorTable", () => {
+  const actors: { [actorId: string]: Actor } = {
+    ACTOR_1: {
+      actorId: "ACTOR_1",
+      jobId: "01000000",
+      address: {
+        rayletId: "426854e68e4225b3941deaf03c8dcfcb1daacc69a92711d370dbb0e1",
+        ipAddress: "172.31.11.178",
+        port: 10003,
+        workerId: "b8b276a03612644098ed7a929c3b0e50f5bde894eb0d8cab288fbb6d",
+      },
+      state: "ALIVE",
+      numRestarts: "0",
+      name: "",
+      pid: 25321,
+      startTime: 1679010689148,
+      endTime: 0,
+      actorClass: "Counter",
+      exitDetail: "-",
+      requiredResources: {},
+      placementGroupId: "123",
+    },
+    ACTOR_2: {
+      actorId: "ACTOR_2",
+      jobId: "01000000",
+      address: {
+        rayletId: "426854e68e4225b3941deaf03c8dcfcb1daacc69a92711d370dbb0e1",
+        ipAddress: "172.31.11.178",
+        port: 10003,
+        workerId: "b8b276a03612644098ed7a929c3b0e50f5bde894eb0d8cab288fbb6d",
+      },
+      state: "DEAD",
+      numRestarts: "0",
+      name: "",
+      pid: 25322,
+      startTime: 1679010689150,
+      endTime: 0,
+      actorClass: "Counter",
+      exitDetail: "-",
+      requiredResources: {},
+      placementGroupId: "123",
+    },
+  };
+  it("renders a table of actors sorted by state", () => {
+    const actors1 = JSON.parse(JSON.stringify(actors));
+    const { getByRole } = render(
+      <MemoryRouter>
+        <ActorTable actors={actors1} />
+      </MemoryRouter>,
+    );
+    // Check that the table is rendered
+    const table = getByRole("table");
+
+    // Check that the actors are sorted by state and startTime
+    const rows = table.querySelectorAll("tbody tr");
+    const actor1Row = rows[0];
```

Comment:
So using indexing here can feel fragile.

One thing that we can do is something like:
```
const actorRow1 = screen.getByRole("row", {
        name: /actor_name/,
      });
      expect(
        within(actorRow1).textContent,
      ).toContain("ACTOR_1");
```